k8s/src/main/scala/io/buoyant/k8s/Ns.scala

```diff
@@ -15,7 +17,7 @@ abstract class Ns[O <: KubeObject: Manifest, W <: Watch[O]: Manifest, L <: KubeL
   private[this] val caches = Var[Map[String, Cache]](Map.empty[String, Cache])
   // XXX once a namespace is watched, it is watched forever.
   private[this] var _watches = Map.empty[String, Activity[Closable]]
-
+  val noNamespace = "all-namespaces"
```

Comment:
I think this class specifically deals with a namescaped API so I don't think it makes sense to make namespace optional here.